src/main/java/com/exasol/containers/ExasolContainer.java

```diff
@@ -145,6 +147,12 @@ public T withPassword(final String password) {
         return self();
     }
 
+    // [impl->dsn~defining-required-optional-service~1]
```

Comment:
Maybe javadoc is required here for public method. Not sure if sonar already complained about it.

Reply:
Fixed.